modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2SaaSAppTokenRevocationTestCase.java

```diff
@@ -0,0 +1,152 @@
+import org.wso2.carbon.identity.oauth.stub.dto.OAuthConsumerAppDTO;
+import org.wso2.carbon.identity.oauth2.stub.dto.OAuth2TokenValidationRequestDTO;
+import org.wso2.carbon.identity.oauth2.stub.dto.OAuth2TokenValidationRequestDTO_OAuth2AccessToken;
+import org.wso2.carbon.identity.oauth2.stub.dto.OAuth2TokenValidationResponseDTO;
+import org.wso2.identity.integration.common.clients.application.mgt.ApplicationManagementServiceClient;
+import org.wso2.identity.integration.common.clients.oauth.Oauth2TokenValidationClient;
+import org.wso2.identity.integration.common.clients.usermgt.remote.RemoteUserStoreManagerServiceClient;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Test class to test token validation after service provider is SaaS-disabled.
+ */
+public class OAuth2SaaSAppTokenRevocationTestCase extends OAuth2ServiceAbstractIntegrationTest {
+
+    private static final String TOKEN_API_ENDPOINT = "/oauth2/token";
+    private static final String TOKEN_TYPE = "bearer";
+
+    private String consumerKey;
+    private String consumerSecret;
+    private String tenantAccessToken;
+    private String superTenantAccessToken;
+    private String tokenEndpointUrl;
+    private String adminUsername;
+    private String adminPassword;
+    private String tenantAdminUsername;
+    private String tenantAdminPassword;
+    private ApplicationManagementServiceClient applicationManagementServiceClient;
+    private Oauth2TokenValidationClient oauth2TokenValidationClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        AutomationContext tenantContext = new AutomationContext("IDENTITY", TestUserMode.TENANT_ADMIN);
+        tenantContext.getContextTenant().getTenantAdmin().getUserName();
+
+        super.init(TestUserMode.SUPER_TENANT_ADMIN);
+        ConfigurationContext configContext = ConfigurationContextFactory
+                .createConfigurationContextFromFileSystem(null, null);
+        applicationManagementServiceClient =
+                new ApplicationManagementServiceClient(sessionCookie, backendURL, configContext);
+        oauth2TokenValidationClient = new Oauth2TokenValidationClient(backendURL, sessionCookie);
+        tokenEndpointUrl = isServer.getContextUrls().getWebAppURLHttps().concat(TOKEN_API_ENDPOINT);
+        adminUsername = isServer.getSuperTenant().getTenantAdmin().getUserName();
+        adminPassword = isServer.getSuperTenant().getTenantAdmin().getPassword();
+        tenantAdminUsername = tenantContext.getContextTenant().getTenantAdmin().getUserName();
+        tenantAdminPassword = tenantContext.getContextTenant().getTenantAdmin().getPassword();
+        OAuthConsumerAppDTO authConsumerAppDTO = createApplication();
+        consumerKey = authConsumerAppDTO.getOauthConsumerKey();
+        consumerSecret = authConsumerAppDTO.getOauthConsumerSecret();
+        assertEquals(applicationManagementServiceClient.getApplication(SERVICE_PROVIDER_NAME).getApplicationName(),
```

Comment:
assertEquals(applicationManagementServiceClient.getApplication(SERVICE_PROVIDER_NAME).getApplicationName()

Here assert should be done first before invoking authConsumerAppDTO.getOauthConsumerKey(); in order to avoid getting  null exception.